x/staking/abci.go

```diff
@@ -0,0 +1,42 @@
+package staking
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/staking/types"
+)
+
+// BeginBlocker will persist the current header and validator set as a historical entry
+// and prune the oldest entry based on the HistoricalEntries parameter
+func BeginBlocker(ctx sdk.Context, k Keeper) {
+	entryNum := k.HistoricalEntries(ctx)
```

Comment:
Best to move this entire block into a method on the keeper.

1. This keeps the `BeginBlocker` abstraction clean.
2. Improves testability.

Reply:
Open to doing this here, but what i have follows the pattern of `BeginBlockers` in `x/slashing`, `x/supply`, and `x/upgrade`

Reply:
They should be updated as well.

Reply:
What about the EndBlocker? Since that isn't a method on the keeper either?